OpenDreamRuntime/Procs/DMOpcodeHandlers.cs

```diff
@@ -267,6 +267,28 @@ public static ProcStatus Enumerate(DMProcState state) {
             return ProcStatus.Continue;
         }
 
+        public static ProcStatus EnumerateAssoc(DMProcState state) {
+            var enumeratorId = state.ReadInt();
+            var outputRef = state.ReadReference();
+            var assocRef = state.ReadReference();
+            var listRef = state.ReadReference();
+            var jumpToIfFailure = state.ReadInt();
+
+            var enumerator = state.Enumerators[enumeratorId];
+            if (enumerator == null || !enumerator.Enumerate(state, outputRef))
+                state.Jump(jumpToIfFailure);
+            else {
+                var outputVal = state.GetReferenceValue(outputRef);
+                state.GetReferenceValue(assocRef);
+                var listVal = state.GetReferenceValue(listRef);
+                var indexVal = state.GetIndex(listVal, outputVal, state);
+
+                state.AssignReference(assocRef, indexVal);
```

Comment:
`AssignReference()` needs done in the correct order same as `GetReferenceValue()`. You're gonna need to reorder the opcode's arguments to have `assocRef` last.